### PR TITLE
feature/print_function

### LINKED ISSUE #8: generic Print() function for printing parameter values
implement a virtual `Print()` function (or similar name) in the base class, and require every inherited class to  implement a specialised version. boilerplate code will probably suffice in most places; sterile osc calc will need to support a generic number of neutrino states.

### PR DESCRIPTION
**Pull request by cjbacchus**
Add a Print() function to the basic IOscCalc interface, and default implementations for the different types of calc.

Plus a little cleanup of TMD5.h includes

### CODE REVIEW

File: OscLib/IOscCalcSterile.cxx [hunk 1]
```diff
@@ -46,6 +46,29 @@ namespace osc
     assert(false);
   }
 
+  //---------------------------------------------------------------------------
+  void IOscCalcSterile::PrintImpl(int nNus, const std::string& prefix) const
+  {
+    for(int i = 1; i < nNus; ++i){
+      std::cout << prefix << "dmsq" << i << "1 = " << GetDm(i) << " eV^2\n";
```
Comment (vhewes) [inline OscLib/IOscCalcSterile.cxx:53]: Masses want to go from 2 -> N, not 1 -> N-1

File: OscLib/IOscCalcSterile.cxx [hunk 2]
```diff
@@ -46,6 +46,29 @@ namespace osc
...
+    }
+
+    for(int i = 0; i+1 < nNus; ++i){
+      for(int j = i+1; j < nNus; ++j){
+        std::cout << prefix << "theta" << i << j << " = " << GetAngle(i, j) << "\n";
+      }
+    }
```
Comment (vhewes) [inline OscLib/IOscCalcSterile.cxx]: i think we want
```
for (int i = 1; i < nNus; ++) {
  for (int j = i+1; j <= nNus; ++j) {
```
so if `nNus = 4`, we cycle from `1 -> 3` in `i`, and `i+1 -> 4` in `j`

File: OscLib/IOscCalcSterile.cxx [hunk 3]
```diff
@@ -46,6 +46,29 @@ namespace osc
...
+
+    for(int i = 0; i+1 < nNus; ++i){
+      for(int j = i+1; j < nNus; ++j){
+        std::cout << prefix << "delta" << i << j << " = " << GetDelta(i, j) << "\n";
+      }
+    }
```
Comment (vhewes) [inline OscLib/IOscCalcSterile.cxx]: same for the deltas as for the angles above, i think

**Review by vhewes — CHANGES_REQUESTED** [review]
indexing for sterile calc needs to be shifted, but otherwise looks good

File: OscLib/IOscCalcSterile.cxx (hunk 1 above)
Comment (cjbacchus) [reply OscLib/IOscCalcSterile.cxx:53]: :man_facepalming: thanks!

**Review by vhewes — APPROVED** [review]

**Pull request merged**